File: 0506-relative-ranks/0506-relative-ranks.kt
class Solution {
fun findRelativeRanks(score: IntArray): Array<String> {
    val sorted = score.sortedArrayDescending()
    val map = sorted.mapIndexed { index, i -> i to index }.toMap()
    return score.map { map[it]!! }.map {
        when (it) {
            0 -> "Gold Medal"
            1 -> "Silver Medal"
            2 -> "Bronze Medal"
            else -> (it + 1).toString()
        }
    }.toTypedArray()

}
}

File: 0506-relative-ranks/0506-relative-ranks.py
class Solution:
    def findRelativeRanks(self, score: List[int]) -> List[str]:
        s = {n: i for i, n in enumerate(sorted(score, reverse=True))}
        medals = ['Gold', 'Silver', 'Bronze']
        return [str(s[n]+1) if s[n] >= len(medals) else (medals[s[n]] + ' Medal') for n in score]